### ISSUE TITLE
Move add_ssh_keys ahead of call to setup_heroku.sh

### ISSUE THREAD

**Issue by christianklotz** [circleci/circleci-docs#1848]
This issue is about <https://circleci.com/docs/2.0/deployment_integrations/> (source file: <https://github.com/circleci/circleci-docs/blob/master/jekyll/_cci2/deployment_integrations.md>)

Following the instruction I kept running into an issue around `heroku.com` not being a known host.

```
The authenticity of host 'heroku.com ([IP address])' can't be established.
RSA key fingerprint is [key fingerprint].
Are you sure you want to continue connecting (yes/no)?
```

1. The documentation states to add the SSH key for the hostname `git.heroku.com`, however the host given in the message states `heroku.com`. Should this be updated?
2. Even after adding an SSH key for `heroku.com` it wouldn't work. Only once I had moved the `add_ssh_keys` to be run _before_ `bash .circleci/setup-heroku.sh` it would work.

**Comment by levlaz** [comment]
What command are you running that gives you this message?

> The authenticity of host 'heroku.com ([IP address])' can't be established.

**Comment by christianklotz** [comment]
If I remember correctly it's the `git push` that caused it.

```bash
git push --force git@heroku.com:$HEROKU_APP_NAME.git HEAD:refs/heads/master
```

For the sake of completeness here's the full `setup-heroku.sh`

```bash
#!/bin/bash
wget https://cli-assets.heroku.com/heroku-cli/channels/stable/heroku-cli-linux-x64.tar.gz -O heroku.tar.gz
sudo mkdir -p /usr/local/lib/heroku /usr/local/bin
sudo tar -xvzf heroku.tar.gz -C /usr/local/lib/heroku --strip-components=1
sudo ln -s /usr/local/lib/heroku/bin/heroku /usr/local/bin/heroku

cat > ~/.netrc << EOF
machine api.heroku.com
  login $HEROKU_LOGIN
  password $HEROKU_API_KEY
EOF

cat >> ~/.ssh/config << EOF
  VerifyHostKeyDNS yes
  StrictHostKeyChecking no
EOF

heroku container:login
heroku plugins:install heroku-repo # required to flush all caches
```

**Comment by smart-alek** [comment]
@christianklotz just confirmed the same issue. Looks like the problem was introduced when we updated the recommended deploy step for Heroku. I'll keep you posted, and thanks for raising this!

**Comment by DrSensor** [comment]
Still bumping to this issue when pushing to github pages
```yml
        - run:
            name: Deploy to org-name.github.io
            command: |
              cd .vuepress/dist
              git init
              git add -A
              git commit -m 'deploy'
              git push -f git@github.com:org-name/org-name.github.io.git master
```

**Comment by smart-alek** [comment]
Hey @DrSensor — thanks for posting. This issue was related to documentation on how to deploy to Heroku. Were you following any of those articles for deploying to GitHub Pages?

**Comment by smart-alek** [comment]
@DrSensor I'm closing this it's not directly related to the original Heroku issue. If you're still having problems with deploying to GitHub Pages, head to our [Support Center](https://support.circleci.com/hc/en-us), or you can always make a new issue. Thanks!
